conda_self/package_info.py

```diff
@@ -26,17 +30,19 @@ def __init__(self, dist_info_path: Path):
         self.dist_info_path = dist_info_path
 
     @classmethod
-    def from_conda_extracted_package_path(cls, path: str | Path):
+    def from_record(cls, record: PackageCacheRecord) -> list[PackageInfo]:
+        return cls.from_conda_extracted_package_path(record.extracted_package_dir)
+
+    @classmethod
+    def from_conda_extracted_package_path(cls, path: str | Path) -> list[PackageInfo]:
         """Create a PackageInfo object given the path to an extracted conda package"""
         path = Path(path)
         matching_paths = [
             p for p in path.rglob("**/*site-packages/*dist-info*") if p.is_dir()
         ]
-        if len(matching_paths) > 1:
-            raise MultipleDistInfoDirsFound()
-        elif len(matching_paths) == 0:
-            raise NoDistInfoDirFound()
-        return cls(matching_paths[0])
+        if len(matching_paths) == 0:
+            raise NoDistInfoDirFound(f"No *.dist-info directories found in {path}")
```

Comment:
While we are cleaning up these exceptions, can we also:
- remove the unused `MultipleDistInfoDirsFound` 
- move `NoDistInfoDirFound` into the exceptions module